core/src/smartcontracts/isi/mod.rs

```diff
@@ -213,14 +211,14 @@ pub mod error {
     impl From<FixedPointOperationError> for Error {
         fn from(err: FixedPointOperationError) -> Self {
             match err {
-                FixedPointOperationError::NegativeValue(_) => Self::Math(MathError::NegativeValue),
```

Comment:
On this occasion only, I would prefer remaining explicit and not relying on the `From` implementations generated by `FromVariant`.